RealmSwift/Util.swift

```diff
@@ -75,18 +75,37 @@ extension ObjectBase {
     }
 }
 
+internal func coerceToNil(_ value: Any) -> Any? {
+    if value is NSNull {
+        return nil
+    }
+    // nil in Any is bridged to obj-c as NSNull. In the obj-c code we usually
+    // convert NSNull back to nil, which ends up as Optional<Any>.none
+    if case Optional<Any>.none = value {
+        return nil
+    }
+    return value
+}
+
 // MARK: CustomObjectiveCBridgeable
 
 /// :nodoc:
 public func dynamicBridgeCast<T>(fromObjectiveC x: Any) -> T {
-    if T.self == DynamicObject.self {
+    return failableDynamicBridgeCast(fromObjectiveC: x)!
+}
+
+/// :nodoc:
+public func failableDynamicBridgeCast<T>(fromObjectiveC x: Any) -> T? {
```

Comment:
should this and `dynamicBridgeCast` not be prefixed with `_`

Reply:
I'm not sure why dynamicBridgeCast is public at all. I'll make the new function internal, though.